### PR TITLE
Update the "new release" issue template

### PR DESCRIPTION
**Pull request by mike-myers-tob**
This PR just updates the "New Release" issue template, specifically because we no longer post new tagged release binaries to Slack. They become available on the Releases page as build artifacts.

That change was mine. Then GitHub has updated the other templates with some additional hidden fields.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/New_Release.md [hunk 1]
```diff
@@ -12,13 +16,13 @@ about: Checklist of Release actions
 ## Before creating a GitHub tag
 
 - [ ] Review the [milestones](https://github.com/osquery/osquery/milestones), move unfinished issues to the next release, close the milestone.
-- [ ] Update the `CHANGELOG.md`, review and merge the change.
-- [ ] Assure that testable packages are available for the `#core` channel on Slack.
-- [ ] Ask for a volunteer to make a GitHub tag.
-- [ ] Assure that the ChangeLog shows up in the [tag release notes](https://github.com/osquery/osquery/tags).
+- [ ] Ask for a Technical Steering Committee member to make a GitHub tag.
+- [ ] Assure that testable packages are available on the Releases page, and announce to the `#core` channel on Slack.
```
Comment (directionless) [inline .github/ISSUE_TEMPLATE/New_Release.md:20]:
```suggestion
- [ ] Ask for a Technical Steering Committee member to make a GitHub tag.
- [ ] Trigger [code signing workflow](https://github.com/osquery/osquery-codesign/actions/workflows/release-generator.yml)
- [ ] Assure that testable packages are available on the Releases page, and announce to the `#core` channel on Slack.
```

File: .github/ISSUE_TEMPLATE/New_Release.md [hunk 2]
```diff
@@ -12,13 +16,13 @@ about: Checklist of Release actions
...
 
 ## After creating a GitHub tag
 
+- [ ] If the `CHANGELOG.md` hasn't already been updated, update it in a new Pull Request, review and merge it. It should reflect everything done up to the tagged version commit.
+- [ ] Assure that the ChangeLog shows up in the [tag release notes](https://github.com/osquery/osquery/tags).
 - [ ] Create packages and upload to S3 (for download use only).
```
Comment (directionless) [inline .github/ISSUE_TEMPLATE/New_Release.md:26]:
```suggestion
- [ ] Assure that the ChangeLog shows up in the [tag release notes](https://github.com/osquery/osquery/tags).

## Promoting to Stable
- [ ] Use [code signing workflow](https://github.com/osquery/osquery-codesign/actions/workflows/release-generator.yml) to build packages and upload to S3
```

**Review by directionless — COMMENTED** [review]
Honestly, it seems awkward as a template. Maybe better as a doc. But I think I suggested the correct corrections

**Comment by mike-myers-tob** [comment]
> Honestly, it seems awkward as a template. Maybe better as a doc. But I think I suggested the correct corrections

Good additions, thank you 👍🏻

**Review by directionless — APPROVED** [review]

**Pull request merged**